### PR TITLE
Fixes to new StaticArrays.QR

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

File: src/qr.jl [hunk 1]
```diff
@@ -45,12 +59,12 @@ _qreltype(::Type{T}) where T = typeof(zero(T)/sqrt(abs2(one(T))))
     SizeQ = Size( sA[1], diagsize(Size(A)) )
     SizeR = Size( diagsize(Size(A)), sA[2] )
 
-    if pivot == Val(true)
+    if pivot === Val{true}
```
Comment (fredrikekre) [inline src/qr.jl]: We are in type-domain here so we never hitted this branch... :)

File: test/qr.jl [hunk 2]
```diff
@@ -27,20 +27,19 @@ srand(42)
...
-        # # pivot=true cases are not released yet
-        # pivot = Val(true)
-        # QRp = @inferred qr(arr, pivot)
-        # @test QRp isa Tuple
-        # @test length(QRp) == 3
-        # Q, R, p = QRp
-        # @test Q isa StaticMatrix
-        # @test R isa StaticMatrix
-        # @test p isa StaticVector
+        # pivot=true cases are not released yet
```
Comment (tkoolen) [inline test/qr.jl]: Is this comment true/helpful at this point?

Comment (fredrikekre) [reply test/qr.jl]: I don't know what it means -- perhaps that there is not released StaticArray-specific version and that we just call to LAPACK? I can add that instead

Comment (tkoolen) [reply test/qr.jl]: I don't know either. Looks like it was added by @denius in https://github.com/JuliaArrays/StaticArrays.jl/pull/328. Probably your interpretation is right.

**Comment by fredrikekre** [comment]
Failures should be fixed by https://github.com/JuliaLang/julia/pull/27862, note that the failures are in the test code where we compare with regular Arrays, and everything still works for StaticArrays.

**Comment by fredrikekre** [comment]
Good to go? Would be good to merge this and retag https://github.com/JuliaLang/METADATA.jl/pull/15400

**Pull request merged**